File: 집합.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <string>

using namespace std;

int main(){
  ios::sync_with_stdio(false);
  cin.tie(0);
  int m ,n; 
  string cmd;
  cin>>m;
  int s=0;
  for(int i=0; i<m; i++){
    cin>>cmd;
    if(cmd=="add"){
      cin>>n; n-=1;
      s |= (1<<n);
    }else if(cmd=="remove"){
      cin>>n; n-=1;
      s &= ~(1<<n);
    }else if(cmd=="check"){
      cin>>n; n-=1;
      int ret = s & 1<<n;
      if(ret) cout<<"1\n";
      else cout<<"0\n";
    }else if(cmd=="toggle"){
      cin>>n; n-=1;
      s ^= 1<<n;
    }else if(cmd=="all"){
      s= (1<<20) -1;
    }else if ( cmd=="empty"){
      s=0;
    }
  }
  return 0;
}

File: 집합.py
import sys 
input =sys.stdin.readline
print=sys.stdout.write
m=int(input())
s=0
for _ in range(m):
  cmd,*num =input().split()
  if len(num) >0 : x=int(num[0])-1#비트연산을 위해 1감소
  
  if cmd == "add":
    s |= (1 << x)
  elif cmd == "remove":
    s &= ~(1<<x)
  elif cmd =="check":
    ret = s&(1<<x)
    if ret > 0: print("1\n")
    else : print("0\n")
  elif cmd == "toggle":
    s ^=(1<<x)
  elif cmd=="all":
    s= (1<<20)-1
  elif cmd=="empty":
    s=0
  
